File: backend/src/models/insights.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel

InsightKind = Literal["decision", "action", "risk", "open_question"]


class Decision(BaseModel):
    id: str
    meeting_id: str
    content: str
    created_at: datetime


class ActionItem(BaseModel):
    id: str
    meeting_id: str
    content: str
    owner: str | None = None
    due_date: datetime | None = None
    created_at: datetime


class Risk(BaseModel):
    id: str
    meeting_id: str
    content: str
    created_at: datetime


class OpenQuestion(BaseModel):
    id: str
    meeting_id: str
    content: str
    created_at: datetime

File: frontend/src/types/insights.ts
export interface Decision {
  id: string
  meeting_id: string
  content: string
  created_at: string
}

export interface ActionItem {
  id: string
  meeting_id: string
  content: string
  owner?: string | null
  due_date?: string | null
  created_at: string
}

export interface Risk {
  id: string
  meeting_id: string
  content: string
  created_at: string
}

export interface OpenQuestion {
  id: string
  meeting_id: string
  content: string
  created_at: string
}
